content/en/posts/js/introduction-a-reactjs/index.md

```diff
@@ -0,0 +1,176 @@
+---
+date: "2014-06-04"
+route: /posts/js/introduction-a-reactjs/
+title: Introducing ReactJS
+tags:
+  - javascript
+  - reactjs
+authors:
+  - bloodyowl
+  - skinnyfoetusboy
+---
+
+It's been quite some time since ReactJS has been open-sourced by facebook.
+Let's take it as an opportunity to introduce the last-born child of the MV* trend.
+
+## Different takes on UI
+
+### Backbone
+
+[Backbone](http://backbonejs.org) brings a simplification in events declaration,
+but doesn't indulge quite that much in rendering.
+Making the developer the sole master of `Backbone.View`
+is positive for several kinds of projects, but makes
+DOM manipulation tedious.
+
+On the subject of data-binding, `Backbone` doesn't actually come
+with any in-house solutions and you're stuck with using mixins to
+implement it without resorting to gnawing your own legs off.
+
+Moreover, it's usually a templating engine ala Handlebars or Jade
+that creates the original view. Thus, you find yourself with some sort
+of `this.$el.html(this.template(data))` in your `render()` method which
+will leave you with a few UX aberrations like your pictures re-rendering
+themselves or videos playing back from the start.
+
+In a nutshell, Backbone is pretty useful to create some sort of
+structure your code, but it is absolutely lackluster in
+diminishing DOM-related complexity concerning the UI of your app.
+
+### Angular
+
+[Angular](https://angularjs.org) brings a more in-depth approach
+by imposing its own templating engine (which allows preprocessing)
+and making you able to declare data-bindings with a `{{mustache}}`-like
+syntax.
+
+Events are declarated in much the same way, using `ng-{eventName}` attributes.
+
+Basically, Angular is technically really interesting (but don't get me started
+on dirty checking)
+
+### Ember
+
+[Ember](http://emberjs.com) is a well-thought and well-developed framework.
+It integrates `data-binding` thanks to [DOM Ranges](https://developer.mozilla.org/en-US/docs/Web/API/range).
+It also has strong design patterns and contrary to what people commonly
+think, it's actually pretty straightforwad.
+Subviews are also pretty easy to use thanks to `{{outlet}}`.
+
+In another words, Ember and Angular both come bundled with real
+solutions for UI management. But they also share the same way to
+do it :
+
+- _rendering_ (creating and inserting DOM nodes) once
+- updating the bindings
```

Comment:
that is no longer true for ember, but we'll keep the original date for people not to get confused